### DISCUSSION TITLE
Need help setting foreground color

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [tui-cs/Terminal.Gui#2665] [General]
I'm not sure what I'm doing wrong, or perhaps I'm misunderstanding the documentation. I'm doing this in PowerShell in PowerShell 7 and it mostly works. I can define a textview in a frame like this.

```powershell
$txtDetail = [TextView]::New()
$txtDetail.Text = 'Process Details'
$txtDetail.ReadOnly = $True
$txtDetail.x = 1
$txtDetail.Width = [Dim]::Fill()
$txtDetail.Height = [Dim]::Fill()

$n = [Terminal.Gui.Attribute]::new("BrightGreen","magenta")
$cs = [Terminal.Gui.ColorScheme]::new()
$cs.focus = $n
$TxtDetail.ColorScheme = $cs
$Frame2.Add($txtDetail)
```

I can get the background in the right-hand frame to change.

![image](https://github.com/gui-cs/Terminal.Gui/assets/12089920/53c15e03-e0ff-4748-a97c-e4c6bd3724b4)

Is there something I'm missing to get the foreground color applied? Or how would you do it from C#?

**Comment by jdhitsolutions** [comment]
As far as I can tell, the color scheme is properly defined.

```powershell
Value          : 90
Foreground     : BrightGreen
Background     : Magenta
Initialized    : True
HasValidColors : True
```

But it doesn't render. I'm running this in Windows Terminal.

**Comment by BDisp** [comment]
The ColorScheme class has various Attributes (Normal, Focus, HotNormal, HotFocus, Disabled). You only defined the Focus attribute and the others aren't valid.

**Comment by jdhitsolutions** [comment]
So I set all attributes to use the same color scheme? I don't think I'm completely understanding the different attributes.

**Reply by BDisp** [reply]
ColorScheme isn't a unique color attribute setting. As I said it has a set of attributes for when haven't focus (Normal), when have focus (Focus) etc. I see the TextView has not the focus and perhaps is using the magenta and white color (default). Set at least the Normal attribute.

**Comment by tznind** [comment]
Hi @jdhitsolutions thanks reaching out.  `ColorScheme` defines the colors (`Attribute`) which should be used in each of the given states a `View` can be in.  But if you are setting them all to the same attribute then it should just work.

Here is the C# that results in something similar to your screenshot.  Hopefully it will give you some insight into your problem.

![image](https://github.com/gui-cs/Terminal.Gui/assets/31306100/d69c9ea2-7a0f-4b00-9536-0de185436dd7)

```csharp
using System.Text;
using Terminal.Gui;

Application.Init();

var w = new Window();
var left = new FrameView() { Width = Dim.Percent(50),Height = Dim.Fill(), Title = "Process List" };
var right = new FrameView() { Width = Dim.Percent(50), Height = Dim.Fill(), Title = "Process Details", X = Pos.Right(left) };

var allGreen = new ColorScheme
{
    Normal = new Terminal.Gui.Attribute(Color.BrightGreen, Color.Magenta),
    Disabled = new Terminal.Gui.Attribute(Color.BrightGreen, Color.Magenta),
    Focus = new Terminal.Gui.Attribute(Color.BrightGreen, Color.Magenta),
    HotFocus = new Terminal.Gui.Attribute(Color.BrightGreen, Color.Magenta),
    HotNormal = new Terminal.Gui.Attribute(Color.BrightGreen, Color.Magenta),
};

var txtDetail = new TextView()
{
    Text = @"This is a test of the emergency
broadcast system.
Zombies are approaching, flee fast!",
    ColorScheme = allGreen,
    Width = Dim.Fill(),
    Height = Dim.Fill(),
    ReadOnly = true
};

right.Add(txtDetail);

var listView = new ListView(new List<string> { "Cat", "Dog", "conhost" })
{
    Width = Dim.Fill(),
    Height = Dim.Fill(),
};

left.Add(listView);

w.Add(left);
w.Add(right);

Application.Run(w);
Application.Shutdown();
```

**Comment by jdhitsolutions** [comment]
I found that if I set the Focus and Disabled attributes to my color scheme, I get the desired result. Not intuitive, but I'm used to hacking my way to the end. Thanks for the comments.

**Comment by BDisp** [comment]
You welcome. I think set the Focus and the Normal you would got the same result. That the way how `ColorScheme` works which is more complex than only set only one foreground and background.
